### DISCUSSION TITLE
How to use findUnique with fields not connected by FK (but they're kinda connected)

### DISCUSSION THREAD

**Discussion by BJladika** [prisma/prisma#7549] [Q&A] [answered]
Okay, I tried to add another field to gql model,
```graphql
type Student {
  id_People: ID!
  Status: [Status]
}
type Status{
  id_PeoplePln: ID!
  id_People: ID
  Group: Group
  id_Group: ID
}
```
```prisma
model people_Pln {
  id_PeoplePln        Int            @id
  id_People           Int
  id_Group            Int
  pln_group_Stud      pln_group_Stud @relation(fields: [id_Group], references: [id_Group])
}
model people_List {
  id_People           Int              @id
  Surname             String           @db.VarChar(30)
  Name                String           @db.VarChar(30)
  Patronymic          String?          @db.VarChar(30)
  id_Nationality      Int?
  spr_Nationality     spr_Nationality? @relation(fields: [id_Nationality], references: [id_Nationality])
}
```
So, I m wrote this
```ts
    Status: (parent, _args, ctx) => {
      return ctx.prisma.people_Pln.findUnique({where: {id_People: parent.id_People}})
    }
```
and trying to call
```graphql
query{
  students{
    Status{
      id_People
      id_Group
    }
  }
}
```
and receiving
```
"message": "\nInvalid `prisma.people_Pln.findUnique()` invocation:\n\n{\n  where: {\n    id_People: 53823\n    ~~~~~~~~~\n  
}\n}\n\nUnknown arg `id_People` in where.id_People for type people_PlnWhereUniqueInput. Did you mean `id_PeoplePln`? Available 
args:\ntype people_PlnWhereUniqueInput {\n  id_PeoplePln?: Int\n}\n\n"
```
kinda worked with usual
```ts
return ctx.prisma.people_Pln.findMany({where: {id_People: parent.id_People}})
```
but its kinda causes N+1 query which is not good
So... what I'm doing wrong now?

**Comment by ryands17** [comment, marked as answer]
`findUnique` only works on an `@id` or a `unique` field. You will have to use either `findMany` or `findFirst` or query with a unique field.

**Reply by BJladika** [reply]
So I'm doomed to N+1 queries?

**Reply by ryands17** [reply]
Currently yes. Either that or use a unique field to query with `findUnique`. We have an issue here for `findMany` that you can follow for more updates: https://github.com/prisma/prisma/issues/1477

**Reply by BJladika** [reply]
Ah, you mentioned making unique field, so in short I've added `@unique` to `id_People           Int`  and now I can make findUnique which was mentioned... okay then...

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
